lib/kramdown/parser/gfm.rb

```diff
@@ -154,6 +154,31 @@ def parse_strikethrough_gfm
         el
       end
 
+      # To handle task-lists we override the parse method for lists, converting matching text into checkbox input
+      # elements where necessary (as well as applying classes to the ul/ol and li elements).
+      def parse_list
+        super
+        @tree.children.each do |element|
+          if [:ul, :ol].include? element.type
+            is_tasklist = false
+            element.children.each do |li|
+              # li -> p -> raw_text
+              checked = li.children[0].children[0].value.gsub!(
+                  /\[ \]\s+/, '<input type="checkbox" class="task-list-item-checkbox" disabled="disabled" />')
+              unchecked = li.children[0].children[0].value.gsub!(
+                  /\[x\]\s+/i, '<input type="checkbox" class="task-list-item-checkbox" disabled="disabled" checked="checked" />')
+              is_tasklist = (checked != nil or unchecked != nil)
```

Comment:
Please use `||` instead of `or`; and `checked.nil?` as this avoids a method call.

Since `is_tasklist` is also used outside this loop, you need to use `is_tasklist ||= ...` since otherwise only the state of the last list item determines the status (which is probably not wanted).